### PR TITLE
Pipeline / pgcraft API improvements

### PR DESCRIPTION
**Pull request by mbouron**

### CODE REVIEW

File: libnopegl/src/pgcraft.c [hunk 1]
```diff
@@ -1552,6 +1552,19 @@ const struct darray *ngli_pgcraft_get_texture_infos(const struct pgcraft *s)
     return &s->texture_infos;
 }
 
+int32_t ngli_pgcraft_get_image_index(const struct pgcraft *s, const char *name)
```
Comment (ubitux) [inline libnopegl/src/pgcraft.c:1557]: no need for the stage check like the other finders?

Comment (mbouron) [reply libnopegl/src/pgcraft.c:1557]: I don't think we support registering 2 pgcraft_textures with the same name to different stages. That would end up in a collision between some uniforms.

Comment (mbouron) [reply libnopegl/src/pgcraft.c:1557]: This is what happens currently on main:
``` diff
--- a/tests/texture.py
+++ b/tests/texture.py
@@ -387,6 +387,7 @@ def texture_2d_array(cfg: ngl.SceneCfg):
     program = ngl.Program(vertex=_TEXTURE2D_ARRAY_VERT, fragment=_TEXTURE2D_ARRAY_FRAG)
     program.update_vert_out_vars(var_tex0_coord=ngl.IOVec2())
     render = ngl.Render(quad, program)
+    render.update_vert_resources(tex0=texture)
     render.update_frag_resources(tex0=texture)
     return render
```
->
```
[ERROR] src/backends/gl/program_gl.c:72 program_check_status: could not compile shader: 0:0(0): error: `tex0_coord_matrix' redeclared
[ERROR] src/backends/gl/program_gl.c:293 ngli_program_gl_init: failed to compile shader "program":
 1 #version 460
 2
 3 #define ngl_out_pos gl_Position
 4 #define ngl_vertex_index gl_VertexID
 5 #define ngl_instance_index gl_InstanceID
 6 layout(location=0) out  vec2 var_tex0_coord;
 7 layout(binding=0) uniform  sampler2DArray tex0;
 8 layout(location=0) in  vec3 ngl_position;
 9 layout(location=1) in  vec2 ngl_uvcoord;
10 layout(location=2) in  vec3 ngl_normal;
11 layout(std140,binding=0) uniform ngl_vert_block {
12     mat4 ngl_modelview_matrix;
13     mat4 ngl_projection_matrix;
14     mat3 ngl_normal_matrix;
15     mat4 tex0_coord_matrix;
16     mat4 tex0_coord_matrix;
17 } ;
18
19 void main()
20 {
21     ngl_out_pos = ngl_projection_matrix * ngl_modelview_matrix * vec4(ngl_position, 1.0);
22     var_tex0_coord = (tex0_coord_matrix * vec4(ngl_uvcoord, 0.0, 1.0)).xy;
23 }
[ERROR] src/nodes.c:604 ngli_node_prepare: preparing node texture_2d_array failed: invalid input data
```

Comment (ubitux) [reply libnopegl/src/pgcraft.c:1557]: sad. thanks.

**Review by ubitux — APPROVED** [review]

**Pull request merged**